src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -188,6 +188,64 @@ static int xocl_mmap(struct file *filp, struct vm_area_struct *vma)
 	return xocl_native_mmap(filp, vma);
 }
 
+static bool is_mem_region_valid(struct xocl_drm *drm_p,
+		struct mem_data *mem_data)
+{
+	struct xocl_dev *xdev = drm_p->xdev;
+	void *blob;
+	int offset;
+	const u64 *prop;
+	u64 start, end, mem_start, mem_end;
+	const char *ipname;
+	int found = 0;
+
+	if (!XOCL_DSA_IS_MPSOC(xdev) && !XOCL_DSA_IS_VERSAL(xdev))
+		return true;
+
+	/* PLRAM does not have to be accessed by PS */
+	if (IS_PLRAM(mem_data->m_tag))
+		return true;
+
+	blob = XDEV(xdev)->fdt_blob;
+	if (!blob)
+		return true;
+
+	for (offset = fdt_next_node(blob, -1, NULL);
```

Comment:
Maybe move this part to xocl_fdt.c? Trying to hide all fdt_* calls in xocl_fdt.c